src/simplnx/Utilities/ThreadedAlgorithm.cpp

```diff
@@ -0,0 +1,66 @@
+//
+// Created by Michael Jackson on 4/24/25.
+//
+
+#include "ThreadedAlgorithm.hpp"
+
+using namespace nx::core;
+
+ThreadedAlgorithm::ThreadedAlgorithm(const IFilter::MessageHandler& mesgHandler, const std::atomic_bool& shouldCancel)
+: m_MessageHandler(mesgHandler)
+, m_ShouldCancel(shouldCancel)
+{
+}
+
+ThreadedAlgorithm::~ThreadedAlgorithm() noexcept = default;
+
+// -----------------------------------------------------------------------------
+const std::atomic_bool& ThreadedAlgorithm::getCancel() const
+{
+  return m_ShouldCancel;
+}
+
+const IFilter::MessageHandler& ThreadedAlgorithm::getMessageHandler() const
+{
+  return m_MessageHandler;
+}
```

Comment:
```suggestion
bool ThreadedAlgorithm::shouldCancel() const
{
  return m_ShouldCancel.load();
}
```
It is not a good to return a reference in parallel. I see the use for providing cancel call without having the threads store the variable, so I have provided a thread-safe solution that uses the proper access pattern for atomic bool and will be not add overhead since bool values are trivially copyable.

See here for simple breakdown: https://stackoverflow.com/questions/15624010/c-shared-mutex-with-return-by-reference

Reply:
In the case of `std::atomic<T>` the `operator T()` is equivalent to calling `load()`. See https://en.cppreference.com/w/cpp/atomic/atomic/operator_T. Also I would say that the cancel variable doesn't need to be in this class anyways since the class is intended to support the messaging aspect.